src/main/java/de/fearnixx/jeak/teamspeak/voice/connection/info/ConfigVoiceConnectionInformation.java

```
@@ -0,0 +1,98 @@
+
+        try {
+            this.configRef.load();
+
+            if (this.configRef.getRoot() == null) {
+                this.configRef.createRoot();
+            }
+
+        } catch (IOException | ParseException e) {
+            throw new IllegalStateException("Could not read voice connection info config!");
+        }
+    }
+
+    private IConfigNode getInfoNode() {
+        final IConfigNode node = configRef.getRoot().getNode(this.identifier);
+
+        if (node.isVirtual()) {
+            node.setMap();
+            save();
+        }
+
+        return node;
+    }
+
+    @Override
+    public void setClientNickname(String clientNickname) {
+        getInfoNode().getNode(NICKNAME_NODE).setString(clientNickname);
+
+        save();
+    }
+
+    @Override
+    public void setLocalIdentity(LocalIdentity localIdentity) {
+
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        try {
+            localIdentity.save(outputStream);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Failed to save identity!");
+        }
+
+        getInfoNode().getNode(IDENTITY_NODE).setString(outputStream.toString());
+
+        save();
+    }
+
+    @Override
+    public String getIdentifier() {
+        return this.identifier;
+    }
+
+    @Override
+    public LocalIdentity getTeamspeakIdentity() {
+        final String identityString = getInfoNode().getNode(IDENTITY_NODE).asString();
+
+        try {
+            return LocalIdentity.read(new ByteArrayInputStream(identityString.getBytes()));
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Failed to load identity!");
+        }
+    }
+
+    @Override
+    public String getClientNickname() {
+        return getInfoNode().getNode(NICKNAME_NODE).asString();
+    }
+
+    private void save() {
+        try {
+            configRef.save();
+        } catch (IOException e) {
+            //Saving next time
```

No problem with swallowing exceptions in this case but please log a message with the exception.